app/Module/GoogleMapsModule.php

```diff
@@ -832,403 +856,217 @@ private function pedigreeMap() {
 		}
 		// display info under map
 		echo '<hr>';
+
 		// print summary statistics
-		if (isset($curgen)) {
-			$total = pow(2, $curgen) - 1;
-			echo '<div>';
-			echo I18N::plural(
-				'%1$s individual displayed, out of the normal total of %2$s, from %3$s generations.',
-				'%1$s individuals displayed, out of the normal total of %2$s, from %3$s generations.',
-				$count,
-				I18N::number($count), I18N::number($total), I18N::number($curgen)
-			);
-			echo '</div>';
-			if ($priv) {
-				echo '<div>' . I18N::plural('%s individual is private.', '%s individuals are private.', $priv, $priv), '</div>';
-			}
-			if ($count + $priv != $total) {
-				if ($miscount == 0) {
-					echo '<div>' . I18N::translate('No ancestors in the database.'), '</div>';
-				} else {
-					echo '<div>' . /* I18N: %1$s is a count of individuals, %2$s is a list of their names */ I18N::plural(
+		$total = count($this->ancestors);
+		echo '<div>';
+		echo I18N::plural(
+			'%1$s individual displayed, out of the normal total of %2$s, from %3$s generations.',
+			'%1$s individuals displayed, out of the normal total of %2$s, from %3$s generations.',
+			$count,
+			I18N::number($count), I18N::number($total), I18N::number($generations)
+		);
+		echo '</div>';
+		if ($priv) {
+			echo '<div>' . I18N::plural('%s individual is private.', '%s individuals are private.', $priv, $priv), '</div>';
+		}
+
+		if ($count + $priv != $total) {
+			$miscount = count($missing);
+			if ($miscount == 0) {
+				echo '<div>' . I18N::translate('No ancestors in the database.'), '</div>';
+			} else {
+				echo '<div>' . /* I18N: %1$s is a count of individuals, %2$s is a list of their names */
+					I18N::plural(
 						'%1$s individual is missing birthplace map coordinates: %2$s.',
 						'%1$s individuals are missing birthplace map coordinates: %2$s.',
 						$miscount, I18N::number($miscount), implode(I18N::$list_separator, $missing)),
-						'</div>';
-				}
+				'</div>';
 			}
 		}
+
 		echo '</div>';
 		echo '</div>';
-		echo '<script src="', $this->googleMapsScript(), '"></script>';
+
+		foreach ($this->loadScripts(true) as $script) {
+			$controller->addExternalJavascript($script);
+		}
+		$controller->addInlineJavascript($this->pedigreeMapJavascript($marker_data, $link_lines));
 	}
 
 	/**
 	 * Create the Javascript to activate the map.
 	 *
+	 * @param $marker_data[]
+	 * @param $link_lines[]
+	 *
 	 * @return string
 	 */
-	private function pedigreeMapJavascript() {
-
-		$js = '
-		// this variable will collect the html which will eventually be placed in the side bar
-		var gm_ancestors_html = "";
-		// arrays to hold copies of the markers and html used by the side bar
-		// because the function closure trick doesnt work there
-		var gmarkers = [];
-		var index = 0;
-		var lastlinkid;
+	private function pedigreeMapJavascript($marker_data = array(), $link_lines = array()) {
```

Comment:
The parameters are always provided, so don't need the default `= array()`.
Add parameter types to PHPDoc.